### PR TITLE
F3-5443: Add pause before setting the exposure level

### PR DESCRIPTION
**Pull request by hjoe72**
- Add pause before setting the exposure level during the initial manual exposure cycle.
- Do the camera exposure cycling in a thread
- Unblock image publishing during the wait between camera setting changes

### CODE REVIEW

**Review by nardavin — APPROVED** [review]
Seems fine to me. What would it take to also detect during runtime exposure is incorrect and reset on the fly?

**Comment by hjoe72** [comment]
> Seems fine to me. What would it take to also detect during runtime exposure is incorrect and reset on the fly?

yeah I have a separate task for that, more like doing our own auto exposure during runtime

File: nodes/usb_cam_node.cpp [hunk 1]
```diff
@@ -362,15 +362,29 @@ class UsbCamNode
 
   void resetExposureSettings()
   {
+    // Set auto exposure to on
     cam_.is_changing_config(true);
     cam_.set_v4l_parameter(
       "auto_exposure",
       AUTO_EXPOSURE_APERTURE_PRIORITY_MODE
     );
+    cam_.is_changing_config(false);
```
Comment (zzv2) [inline nodes/usb_cam_node.cpp:371]: put this in the `set_v4l_parameter` function at beginning and end

**Review by zzv2 — CHANGES_REQUESTED** [review]

**Comment by zzv2** [comment]
> > Seems fine to me. What would it take to also detect during runtime exposure is incorrect and reset on the fly?
> 
> yeah I have a separate task for that, more like doing our own auto exposure during runtime

I'm weary of any autoexposure. please review the algorithm design with me before implementing

File: nodes/usb_cam_node.cpp (hunk 1 above)
Comment (hjoe72) [reply nodes/usb_cam_node.cpp:371]: done

**Pull request merged**